src/admin.c

```diff
@@ -1697,6 +1739,71 @@ static bool admin_parse_query(PgSocket *admin, const char *q)
 	goto done;
 }
 
+/*
+ * Implements a minimal requirements for supporting extended query protocol for the
+ * admin console. The procedures generated by parse command are not stored, but only
+ * the query is captured within the admin socket. Once the Execute packet is issued,
+ * we will invoke the `admin_parse_query` similar to how we perform the simple protocol
+ * dance.
+ */
+static bool admin_handle_extended_protocol(PgSocket *admin, PktHdr *pkt) {
+	PgBindPacket bp;
+	PgDescribePacket dp;
+	bool res;
+
+	switch (pkt->type) {
+	default:
+		admin_error(admin, "extended query protocol not supported by admin console");
+		disconnect_client(admin, true, "bad packet");
+		return false;
+	case 'P':
+	   	/*
+		 * The Parse packet contains the query, which is the only thing we care in
+		 * this protocol dance.
+		 */
+	  	admin->admin_extended_protocol_cmd = malloc(sizeof(PgParsePacket));
+		if (!admin->admin_extended_protocol_cmd) {
+			admin_error(admin, "failed to allocate parse packet");
+			disconnect_client(admin, true, "out of memory");
+			return false;
+		}
+
+	 	if (!unmarshall_parse_packet(admin, pkt, admin->admin_extended_protocol_cmd)) 
+			return false;
+		
```

Comment:
should check if the statement has a name, and fail if that's the case. Let's not start supporting named prepared statements unless really necessary.